test/e2e/baseTest.ts

```diff
@@ -0,0 +1,82 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import type { Page } from '@playwright/test';
+import { _electron, test as base } from '@playwright/test';
+import { downloadAndUnzipVSCode } from '@vscode/test-electron/out/download';
+
+export { expect } from '@playwright/test';
+
+export type TestOptions = {
+	vscodeVersion: string;
+};
+
+type TestFixtures = TestOptions & {
+	page: Page;
+	createTmpDir: () => Promise<string>;
+};
+
+export const MaxTimeout = 10000;
+
+let testProjectPath: string;
+export const test = base.extend<TestFixtures>({
+	vscodeVersion: ['insiders', { option: true }],
+	page: async ({ vscodeVersion, createTmpDir }, use) => {
+		const defaultCachePath = await createTmpDir();
+		const vscodePath = await downloadAndUnzipVSCode(vscodeVersion);
+		testProjectPath = path.join(__dirname, '..', '..', '..');
+
+		const electronApp = await _electron.launch({
+			executablePath: vscodePath,
+			// Got it from https://github.com/microsoft/vscode-test/blob/0ec222ef170e102244569064a12898fb203e5bb7/lib/runTest.ts#L126-L160
+			args: [
+				'--no-sandbox', // https://github.com/microsoft/vscode/issues/84238
+				'--disable-gpu-sandbox', // https://github.com/microsoft/vscode-test/issues/221
+				'--disable-updates', // https://github.com/microsoft/vscode-test/issues/120
+				'--skip-welcome',
+				'--skip-release-notes',
+				'--disable-workspace-trust',
+				`--extensionDevelopmentPath=${path.join(__dirname, '..', '..', '..')}`,
+				`--extensions-dir=${path.join(defaultCachePath, 'extensions')}`,
+				`--user-data-dir=${path.join(defaultCachePath, 'user-data')}`,
+				testProjectPath,
+			],
+		});
+
+		const page = await electronApp.firstWindow();
+		/*await page.context().tracing.start({
```

Comment:
Do we need commented code here and other places?

Reply:
indeed, good shout. I commented it out, as this piece of code is used to debug (it creates a trace to check the tests step by step). I'll clean the file and remove it